proto/cosmos/tx/v1beta1/tx.proto

```diff
@@ -174,4 +174,9 @@ message Fee {
   // the payer must be a tx signer (and thus have signed this field in AuthInfo).
   // setting this field does *not* change the ordering of required signers for the transaction.
   string payer = 3;
+
+  // if set, the fee payer (either the first signer or the value of the payer field) requests that a fee grant be used
+  // to pay fees instead of the fee payer's own balance. If an appropriate fee grant does not exist or the chain does
+  // not support fee grants, this will fail
```

Comment:
Just a thought: it feels to me like the beginning of a codesmell to put fields inside Tx that depend on what modules are on chain. Tx stuff should be module-agnostic imo. There's a risk we're opening the door to allow other module-specific fields here in the future, which can get messy. What do you think?

Would proto extensions work here?

Reply:
You mean `TxBody.ExtensionOptions`? We're not using protobuf extensions in general because they're deprecated.

I agree `Tx` should be module agnostic, but there are some pieces that live at the protocol level (i.e. `Tx`) and cross over into modules. Fee grants is I think an example of this. We could put it in `ExtensionOptions`, but from my perspective the intention was to always put it into `Fee` somehow. I actually thought we would just have a `payer` field like in #7384, but @ethanfrey suggested an alternate approach which I'm taking here.

I wonder if @ethanfrey has any thoughts on this actually...

Reply:
> You mean TxBody.ExtensionOptions

Yes sorry that's what I meant.

Reply:
I think it is good to add this field here.

The reason for my suggestion is that if there are 3 signers and one wants to pay with a fee grant, we will need 2 fields. One saying who is taking responsibility (payer) and the second whose account it comes from (granter). If we just have granter, we need to check all signers if any have a valid grant

Reply:
Also, I would say there is plenty of x/auth specific data structures in the Tx proto. Fee Grant's are not some generic module, but would be tied as a fundamental part of the tx processing. I guess you could build an sdk app without x/auth, but why?

Just saying, we need to take some opinions to make things usage, like gas and fees, and how to pay fees. Also how to sign. No one has ever deployed a chain that didn't use StdTx, with all the work that went into making it an interface